.github/workflows/package-check.yml

```
@@ -0,0 +1,48 @@
+name: package-check
+
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+
+permissions:
+  contents: read
+
+jobs:
+  package-check:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          persist-credentials: false
+
+      - name: Use Node.js 20.x
+        uses: actions/setup-node@v4.2.0
```

If we're pinning `actions/upload-artifact`, should we pin this as well?

---

```suggestion
        uses: actions/setup-node@cdca7365b2dadb8aad0a33bc7601856ffabcc48e # v4.3.0
```

Additionally, [v4.3.0](https://github.com/actions/setup-node/releases/tag/v4.3.0) is the latest.

---

@joshmgross I was going to but rather chose to keep it the same as all other references in this project for now: https://github.com/tj-actions/changed-files/blob/0fee5fb278312d962ff465bb38dc4cae9f446de2/.github/workflows/test.yml#L47-L51

A pretty good follow-up PR to this one would be going through **all** 3rd party workflows in this project and pinning them to exact commit SHAs IMO.

---

The maintainer seems to have pinned all other actions, I'd suggest pinning here as well.